cmd/graphy/main.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"encoding/csv"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+)
+
+var usage = `Consumes a csv file in the format: YYYY-MM-DD HH:MM:SS,file,timeseries,(start|done)
+and outputs a new csv to stdout that aggregates the time differences between start/end
+
+Usage:
+  graphy [flags] <csvfilename>
+
+Flags:
+      --orientation string   Display orientation: horizontal or vertical (default "horizontal")`
+
+var (
+	orientation = flag.String("orientation", "horizontal", "Control orientation of the data")
+)
+
+func main() {
+	for _, arg := range os.Args {
+		switch arg {
+		case "-h", "--help", "-help":
+			help()
+			return
+		}
+	}
+
+	flag.Parse()
+
+	if err := graphyCmd(flag.Args()); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func help() {
+	fmt.Fprintln(os.Stderr, usage)
+}
+
+func graphyCmd(args []string) error {
+	if len(args) < 1 {
+		help()
+		return errors.New("need more arguments")
+	}
+
+	filename := args[0]
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	events := make(map[string][]uint64)
+	starts := make(map[string]time.Time)
+
+	reader := csv.NewReader(f)
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
```

Comment:
`else if` feels cleaner here. Won't change without good reason. Takes more lines to write the alternative when in reality this block is "checking what err" is so makes sense to me as a single "piece". Could be a switch if you wanna be weird about it.